kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtNetworkServiceImpl.java

```diff
@@ -1237,252 +1046,477 @@ public ArrayList<GwtFirewallNatEntry> findDeviceFirewallNATs(GwtXSRFToken xsrfTo
-                List<NetConfig> netConfigs = new ArrayList<>();
-
-                if (routerMode.equals(GwtNetRouterMode.netRouterDchp.name())
-                        || routerMode.equals(GwtNetRouterMode.netRouterDchpNat.name())) {
-                    int defaultLeaseTime = config.getRouterDhcpDefaultLease();
-                    int maximumLeaseTime = config.getRouterDhcpMaxLease();
-                    IP4Address routerAddress = (IP4Address) IPAddress.parseHostAddress(config.getIpAddress());
-                    IP4Address rangeStart = (IP4Address) IPAddress.parseHostAddress(config.getRouterDhcpBeginAddress());
-                    IP4Address rangeEnd = (IP4Address) IPAddress.parseHostAddress(config.getRouterDhcpEndAddress());
-                    boolean passDns = config.getRouterDnsPass();
-
-                    IP4Address subnetMask = (IP4Address) IPAddress.parseHostAddress(config.getRouterDhcpSubnetMask());
-                    IP4Address subnet = (IP4Address) IPAddress.parseHostAddress(
-                            NetworkUtil.calculateNetwork(config.getIpAddress(), config.getSubnetMask()));
-                    short prefix = NetworkUtil.getNetmaskShortForm(subnetMask.getHostAddress());
-
-                    // Use our IP as the DNS server and we'll use named to proxy DNS queries
-                    List<IP4Address> dnsServers = new ArrayList<>();
-                    dnsServers.add((IP4Address) IPAddress.parseHostAddress(config.getIpAddress()));
-
-                    logger.debug("DhcpServerConfigIP4 - start: {}, end: {}, prefix: {}, subnet: {}, subnetMask: {}",
-                            new Object[] { rangeStart.getHostAddress(), rangeEnd.getHostAddress(), prefix,
-                                    subnet.getHostAddress(), subnetMask.getHostAddress() });
-                    try {
-                        DhcpServerCfg dhcpServerCfg = new DhcpServerCfg(config.getName(), true, defaultLeaseTime,
-                                maximumLeaseTime, passDns);
-                        DhcpServerCfgIP4 dhcpServerCfgIP4 = new DhcpServerCfgIP4(subnet, subnetMask, prefix,
-                                routerAddress, rangeStart, rangeEnd, dnsServers);
-                        netConfigs.add(new DhcpServerConfigIP4(dhcpServerCfg, dhcpServerCfgIP4));
-                    } catch (KuraException e) {
-                        logger.error(
-                                "Failed to create new DhcpServerConfigIP4 object. Please verify that DHCP pool IP addresses (see below) are in the {} subnet.",
-                                subnet.getHostAddress());
-                        logger.error("DHCP Pool: range from {} to {}", rangeStart.getHostAddress(),
-                                rangeEnd.getHostAddress());
-                        logger.error("Exception: ", e);
-                    }
+        try {
+            for (GwtFirewallOpenPortEntry entry : entries) {
+                openPorts.append(entry.getPortRange()).append(",");
```

Comment:
If `entry.getPortRange()` returns `null`, this will probably add the `null` string in the configuration. Is this ok?

Reply:
Mhh... the `null` value is not allowed here. I mean, the web UI forces to set a value in the port range property.